### ISSUE TITLE
--fix mode not fixing anything with eslint-plugin-script-tags.

### ISSUE THREAD

**Issue by trusktr** [eslint/eslint#9142]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 8.1.2
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": [
    "airbnb",
    "prettier",
  ],
  "plugins": [
    "prettier",
    "script-tags",
  ],
  "env": {
    "browser": true,
    "node": true,
    "mocha": true,
  },
  "settings": {
    "import/resolver": "node",
  },
  "rules": {
    "eol-last": 0,
    "prettier/prettier": ["error", {
      "printWidth": 80,
      "tabWidth": 2,
      "useTabs": false,
      "semi": false,
      "singleQuote": true,
      "trailingComma": "all",
      "bracketSpacing": true,
      "jsxBracketSameLine": false,
    }],
    "arrow-parens": 0,
    "no-unused-vars": [
      "error",
      {
        "argsIgnorePattern": "^_",
      },
    ],
    "jsx-a11y/href-no-hash": 0,
    "comma-dangle": [
      "error",
      {
        "arrays": "always-multiline",
        "objects": "always-multiline",
        "imports": "always-multiline",
        "exports": "always-multiline",
        "functions": "never",
      },
    ],
  },
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

I made a reproduction here: https://github.com/trusktr/gulp-eslint/tree/issue-204

Also see https://github.com/adametry/gulp-eslint/issues/204 for more description of the issue.

**What did you expect to happen?**

Basically I expect `--fix` mode to also modify the content of my `<script>` tags because I'm using `eslint-plugin-script-tags` to lint those parts of `.hbs` or `.html` files, etc.

**What actually happened? Please include the actual, raw output from ESLint.**

No fixing happens in `.hbs` file, for example, only in `.js` files.

**Comment by trusktr** [comment]
After running eslint, I get this in my repo:

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   foo.js
        modified:   blah.js
        modified:   bar/index.js
```

but just not `.hbs` files. Still playing with it, trying to figure out why.

**Comment by trusktr** [comment]
I'm using `prettier`, and I _**am**_ getting `prettier` errors for the `.hbs` files, just fixes are not being applied to them.

**Comment by not-an-aardvark** [comment]
Thanks for the report. This is working as intended because `eslint-plugin-script-tags` uses [processors](https://eslint.org/docs/developer-guide/working-with-plugins#processors-in-plugins), and autofixing is currently not supported when using processors (see the [documentation](https://eslint.org/docs/user-guide/command-line-interface#--fix)).

It's possible that this will be supported soon (see https://github.com/eslint/eslint/pull/9090), although it's likely that some changes will be required in `eslint-plugin-script-tags` after that for autofixing to work correctly.

**Comment by trusktr** [comment]
@not-an-aardvark Ah, thanks! So, for people that make processors, what's the way in which they should handle it so I can relay the message back to them?

**Comment by trusktr** [comment]
Here's the [related issue on eslint-plugin-script-tags](https://github.com/mapbox/eslint-plugin-script-tags/issues/6).

**Comment by not-an-aardvark** [comment]
My recommendation to processor developers would be to wait until https://github.com/eslint/eslint/pull/9090 is hopefully merged sometime in the near future, and then update their processors to also transform autofix ranges (as described in https://github.com/eslint/eslint/pull/9090) to make autofixing work properly.

That said, it would also be useful for the ESLint team to get some implementation feedback earlier -- if anyone wants to update a processor to implement the experimental API in https://github.com/eslint/eslint/pull/9090 before it's merged, we'd like to hear about any implementation difficulties/inconveniences you encounter, so that we can modify the API appropriately. However, it's not guaranteed that the current form of the API is the final version that will be shipped (or that this enhancement will be shipped at all), so I would only do this if you're okay with the possibility of needing to change the implementation later on.

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.
